Added BreachSwap module and its API

someRandomDude-a opened the pull request.

someRandomDude-a committed:
Added BreachSwap module and its API

someRandomDude-a committed:
AutoAttack now works well with breachSwap

someRandomDude-a committed:
implementing BeforeAttack event

someRandomDude-a committed:
implement event.cancel()

someRandomDude-a committed:
updated event method

someRandomDude-a committed:
update autoCriticals to use new "BeforeAttack" event

Zergatul commented on common/java/com/zergatul/cheatutils/modules/scripting/EventsScripting.java:
```diff
@@ -107,6 +108,12 @@ private EventsScripting() {
             }
         });
 
+        Events.BeforeAttack.add(event -> {
```
This doesn't do anything, remove

someRandomDude-a replied:
removed

Zergatul commented on common/resources/web/modules.js:
```diff
@@ -88,6 +88,13 @@ module({
+    name: 'Breach Swap',
+    component: 'BreachSwap',
+    path: 'breach-swap',
+    tags: ['auto', 'attack']
```
Update tags

someRandomDude-a replied:
added tags:
    tags: ['auto', 'attack', 'breach', 'mace', 'swap']

Zergatul commented on common/java/com/zergatul/cheatutils/modules/automation/BreachSwap.java:
```diff
@@ -0,0 +1,144 @@
+
+
+    public boolean handling;
+
```
These shouldn't be public

someRandomDude-a replied:
Changed to private, requires testing to make sure it works

Zergatul commented on common/java/com/zergatul/cheatutils/modules/automation/BreachSwap.java:
```diff
@@ -0,0 +1,144 @@
+        }
+    }
+    public boolean attack(boolean useAxe, boolean breakShield) {
+        handling = true;
```
This doesn't do anything, you set handling to true inside run method

someRandomDude-a replied:
This is left over code, removed the references from `run() `
put it into new `attack` method for tidier `handling` state

Zergatul commented on common/java/com/zergatul/cheatutils/mixins/common/MixinMultiPlayerGameMode.java:
```diff
@@ -1,9 +1,11 @@
 import com.zergatul.cheatutils.common.events.PlayerReleaseUsingItemEvent;
 import com.zergatul.cheatutils.configs.ConfigStore;
 import com.zergatul.cheatutils.configs.FastBreakConfig;
+import com.zergatul.cheatutils.scripting.Root;
```
remove this

someRandomDude-a replied:
done

Zergatul commented on common/resources/web/components/automation/BreachSwap.html:
```diff
@@ -0,0 +1,21 @@
+    <div class="description">
+        Breach Swap makes it so that holding the attack button swaps sword and mace to performa Breach Swap attack,
+        Penetrating armor with breach while using base damage and cooldown of a sword or axe.
+        Each swing is done right after the attack cooldown is done and thus will always do max possible damage.
```
Is this really true? I don't see any attack cooldown checks

someRandomDude-a replied:
This is from the previous implementation, should be updated.
now this functionality can be achieved by just using the "Auto Attack" module along with "breach swap" module
instead so removed this from breachSwap module

someRandomDude-a replied:
updated to reflect current implementation

someRandomDude-a committed:
Resolve changes

someRandomDude-a committed:
added skip for auto criticals method

Needed for breach-swap module
Edge case where breachSwap and autoCriticals are enabled ->

if target is using shield, this will result in module attempting to convert 2 attacks in the same tick into criticals
this is not good.

someRandomDude-a committed:
updated handling to static type

Zergatul commented on common/java/com/zergatul/cheatutils/modules/scripting/EventsScripting.java:
```diff
@@ -1,6 +1,7 @@
 package com.zergatul.cheatutils.modules.scripting;
 
 import com.zergatul.cheatutils.common.Events;
+import com.zergatul.cheatutils.common.events.BeforeAttackEvent;
```
Please remove this, so this file does not appear in the list of the changed files

someRandomDude-a committed:
added afterAttack event, removed sprint logic from auto criticals

someRandomDude-a committed:
sprinting logic update

someRandomDude-a committed:
Update AutoCriticals.java

someRandomDude-a committed:
removed unused afterAttack event

someRandomDude-a committed:
updated shield blocking logic
